### DISCUSSION TITLE
Boundary condition type

### DISCUSSION THREAD

**Discussion by AmrIbrahim8894** [idaholab/moose#27410] [Q&A General] [answered]
Hi there, 
Is there a way to get the variable values of the nodes in blue color in the attached picture and impose these values on the boundary nodes in red color for each side of cohesive zone model. Thank you
![Image(1)](https://github.com/idaholab/moose/assets/147007633/deba1a17-d7f7-418a-aeaf-0a81eea8d0a1)

**Comment by GiudGiud** [comment]
Hello

Yes there are a number of ways. Are the meshes stitched in your system or apart but close?

Guillaume

**Comment by AmrIbrahim8894** [comment]
Hello,
I am using BreakMeshByBlockGenerator to create an interface.

**Comment by GiudGiud** [comment]
I think I might have mis-understood. You want the red and blue nodes on both sides of the split, both sides being independent, to have the same values?

**Comment by AmrIbrahim8894** [comment]
yes. both sides are independent, i want to impose the variable values of the blue nodes on the red nodes for each side seperately.

**Reply by GiudGiud** [reply]
if the gradient is 0 that would do get you a flat slope from red to blue. The values would be almost the same

**Reply by GiudGiud** [reply]
you could define a nodeset the blue and for the red nodes, then add a Constraint to force the equality

**Comment by AmrIbrahim8894** [comment]
I am using mixed formulation, so i want to impose the gradient to be zero along with equating the nodes. this will be inside a cohesive zone model that i am trying to develop for a fault embedded in a poroelastodynamic medium. if there is an easier way either than using a constraint, it will be great. thank you

**Comment by AmrIbrahim8894** [comment]
I am using explicit scheme so i cannot use the a constraint like EqualValueBoundaryConstraint. if i can extract the nodesets for both blue and red nodes, how can i equate the variable values for these two nodsets ?

**Comment by AmrIbrahim8894** [comment, marked as answer]
I used `ParsedSubdomainMeshGenerator` to generate new block containing both the blue and red nodes then i used `ParsedGenerateSideset` to create a sideset containing the blue nodes only. Then, I used `NearestNodeValueAux` to save the values of the blue nodes in an auxiliary variable but at the nodal positions of the red nodes. Finally, I imposed the boundary condition using `MatchedValueBC` between the auxiliary variable and main variable. Thank you

**Reply by GiudGiud** [reply]
ok that's pretty clever. Just a note that the contribution to the Jacobian  of the BC is likely not correct from the get go. But if you use PJFNK it does not matter as much as if you were using Newton's method
